src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs

```diff
@@ -204,6 +204,10 @@ internal static int GetWarningLevel(ErrorCode code)
 
             switch (code)
             {
+                case ErrorCode.WRN_AutoPropertyAllowNull:
+                    // Warning level 6 is exclusively for warnings introduced in the compiler
+                    // shipped with dotnet 6 (C# 10) and that can be reported for pre-existing code.
+                    return 6;
```

Comment:
I thought nullable-related warnings are usually okay to be introduced without a warning wave. What's special about this one?

Reply:
I believe that we treated C# 9 as a sort of mop-up release for nullability, where things are allowed to change fairly significantly, but with each subsequent release, the bar for new nullability diagnostics goes up. Therefore it feels like it would be reasonable to group any new diagnostics into warning waves so that users can adopt them gradually if they want.